### PR TITLE
Allow configuration of btrfs with want_rootfs=btrfs

### PR DESCRIPTION
**Pull request by dirkmueller**
There is a new requirement for Cloud7 to enable btrfs by default.
Let's add it as an option until subvolume handling has been implemented.

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -2135,7 +2135,7 @@ function onadmin_allocate()
     fi
 
     # set BTRFS for all nodes when docker is wanted (docker likes btrfs)
-    if [ -n "$want_docker" ] ; then
+    if [ -n "$want_docker" -o -n "$want_btrfs" ] ; then
```
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Please adapt the comment to be non-docker specific

**Comment by bmwiedemann** [comment]
+0.5 we could also do it more generic, by using a want_rootfs=btrfs (e.g. some people were talking about XFS as the recommended alternative to btrfs)
with the default value being empty that leaves crowbar's default.

**Comment by jsuchome** [comment]
+1

**Comment by dirkmueller** [comment]
@bmwiedemann thats an interesting thought.

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -2141,6 +2141,13 @@ function onadmin_allocate()
         done
     fi
 
+    # set rootfs for all nodes when want_rootfs is set
+    if [ -n "$want_rootfs" ] ; then
+        for node in `get_all_discovered_nodes` ; do
+            set_node_fs $node "$want_rootfs"
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: and now the docker case above can be reduced to [ -n "$want_docker" ] && want_rootfs=btrfs

Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: docker works fine without btrfs as well, but it just runs slower then.

Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: with this code you can do want_docker=1 want_rootfs=xfs and it would not deploy docker with btrfs but with xfs.

Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: that could also be done with

```
[ -n "$want_docker" ] && : ${want_rootfs:=btrfs}
```

**Comment by bmwiedemann** [comment]
merged with a small docker change.

**Comment by bmwiedemann** [comment]
caf1555ac225950bfb2e1693876600af6a756904

**Pull request closed without merging**